src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs

```diff
@@ -405,6 +480,35 @@ internal static unsafe int WriteFileNative(SafeFileHandle handle, ReadOnlySpan<b
             }
         }
 
+        private static void GetPathForNtCreateFile(string fullPath, out ReadOnlySpan<char> prefixedAbsolutePath, out char[]? rentedArray)
+        {
+            const string mandatoryNtPrefix = @"\??\";
+
+            if (fullPath.StartsWith(mandatoryNtPrefix, StringComparison.Ordinal))
+            {
+                prefixedAbsolutePath = fullPath;
+                rentedArray = null;
+            }
+            else if (fullPath.StartsWith(@"\\?\", StringComparison.Ordinal)) // NtCreateFile does not support "\\?\" prefix, only "\??\"
+            {
+                rentedArray = ArrayPool<char>.Shared.Rent(fullPath.Length);
```

Comment:
Could stackalloc be used here when the path is smaller than a certain threshold e.g: 100 chars?

Reply:
good point! done